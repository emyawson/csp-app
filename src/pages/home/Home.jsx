/* eslint-disable react/prop-types */
// Modules
import React, { Fragment, useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Layout from '../../components/layout/Layout.jsx';
import Headline from '../../components/headline/Headline.jsx';
import LoadingWrapper from '../../components/loading/LoadingWrapper.jsx';
import BCrumb from '../../components/Bcrumb/BCrumb.jsx';
import Faq from '../../components/faq/Faq.jsx';
import { ROUTE_ERROR_PAGE } from '../../constants/routes';
import TopicsBlock from './TopicsBlock.jsx';

// Selectors
import { selectPages, selectLoadingPages } from '../../store/selectors';

// styles
import './Home.scss';

// eslint-disable-next-line no-unused-vars
const Home = ({ history }) => {
  // Selectors
  const pages = useSelector(selectPages);
  const loadingPages = useSelector(selectLoadingPages);

  const HeadlineLoader = LoadingWrapper(Headline);

  const [homePage, setHomePage] = useState({});
  const [homePageTopics, setHomePageTopics] = useState([]);

  useEffect(() => {
    if (pages) {
      setHomePage(pages);
    }
    if (homePage) {
      setHomePageTopics(homePage.topics);
    }
    if (pages.error !== null) {
      history.push(ROUTE_ERROR_PAGE);
    }
  }, [pages, homePage]);

  return (
    <Layout>
      <BCrumb />
      <Fragment>
        {homePage && <HeadlineLoader
          isLoading={loadingPages}
          title={homePage.headline}
          description={homePage.body}
          loadingMarginTop="60px"
          loadingWidth="505px"
          loadingHeight="106px"
        />}
        <div className="topic-wrapper">
          {homePageTopics && <TopicsBlock homeTopics={homePageTopics} loading={loadingPages} />}
        </div>
        
        <Faq />

      </Fragment>

    </Layout>
  );
};

export default withRouter(Home);
