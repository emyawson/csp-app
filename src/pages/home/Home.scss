@import '../../index';


.topic-wrapper {
  min-height: 236vw;
  @media only screen and (min-width: 429px) {
    min-height: 152vw;
  }
  @include respond(tablet) {
    min-height: 70vw;
  }
  @include respond(desktop) {
    min-height: 34vw;
  }
  @include respond(largeDesktop) {
    min-height: 13vw;
  }
}

.topics-container {
  column-gap: 22px;
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  @include respond(tablet) {
  grid-template-columns: repeat(2, 1fr);
}
@include respond(desktop) {
  column-gap: 30px;
  grid-template-columns: repeat(3, 1fr);
}
@include respond(largeDesktop) {
  grid-template-columns: repeat(4, 1fr);
}
};

.topics-container-additional {
  @extend .topics-container;
  @include respond(largeDesktop) {
  grid-template-columns: repeat(3, 1fr);
}
};
