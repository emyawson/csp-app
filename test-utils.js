import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
