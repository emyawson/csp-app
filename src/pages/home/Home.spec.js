/* eslint-disable no-console */
import React from 'react';
import Home from './Home.jsx';
import { renderWithRouter } from '../../../test-utils';
import pagesMock from '../../services/mock/pagesMock';

const initialState = {
  router: {},
  loading: {},
  pages: pagesMock,
  breadcrumbPathname: [],
  errors: {},
};

const renderComponent = () => renderWithRouter(<Home/>, {}, initialState);

test('renders <Home/ >', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();
});
