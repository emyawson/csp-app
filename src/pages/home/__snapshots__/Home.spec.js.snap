// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Home/ > 1`] = `
<div>
  <main
    class="page-container"
  >
    <div
      class="page-container__inner"
    >
      <div
        class="page-container__main"
      >
        <div
          class="breadcrumb"
        >
          <nav
            aria-label="Breadcrumb"
            class="breadcrumb__nav"
            data-cytest="breadcrumb"
            role="navigation"
          >
            <ol
              class="breadcrumb__list"
              itemprop="breadcrumb"
              itemscope=""
              itemtype="http://schema.org/BreadcrumbList"
            >
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    IKEA
                  </span>
                </a>
                 
                <meta
                  content="1"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    Klantenservice
                  </span>
                </a>
                 
                <meta
                  content="2"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service/support"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    We are here to help and you should be greatful
                  </span>
                </a>
                 
                <meta
                  content="3"
                  itemprop="position"
                />
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="headline"
        >
          <h1
            class="headline__title"
            data-cytest="headline"
          >
            We are here to help and you should be greatful
          </h1>
          <p
            class="headline__description"
          >
            Choose a topic and weâ€™ll find you the best way to get in touch. 

          </p>
        </div>
        <div
          class="topic-wrapper"
        >
          <ul
            class="action-list action-list--regular topics-container-additional"
            data-cytest="topics-list"
          >
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Stores and about IKEA
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Get help finding your closest IKEA store. You can also find out more about IKEA here. 
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    How to shop
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Shopping in store or from home
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Place an order
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Need help placing an order? We can help. 
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Services
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Get help navigating services
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Product & technical information and support
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Here you'll find ways to get help with assembly, buying guides, product recalls and IKEA smart products
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Manage order
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Need to change or edit your order? This is your starting point. 
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Returns, refunds, delivery issues, & missing Items
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Stuff happens, we know. But when it does, you can contact us and get support. 
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Feedback and complaints
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Tell us how we're doing
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="action-list-item"
            >
              <button
                class="action-list-item__action"
              >
                <span
                  class="action-list-item__label-container"
                >
                  <span
                    class="action-list-item__label"
                  >
                    Other help topics
                  </span>
                  <span
                    class="action-list-item__caption"
                  >
                    Some topics might fall between categories. This is where we start. 
                  </span>
                </span>
                <span
                  class="action-list-item__chevron"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </li>
          </ul>
        </div>
        <div
          class="faq"
        >
          <h2
            class="faq__headline"
          >
            Zoeken naar informatie
          </h2>
          <div
            data-cytest="faq"
          >
            <km-search-integration
              lang="nl-NL"
            />
            <km-faq-accordion
              faqlang="nl"
              lazyload="true"
              listtitle=""
              widget="m6"
            />
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
